tests/unit/completion/test_completionwidget.py

```diff
@@ -185,6 +188,36 @@ def test_completion_item_focus_no_model(which, completionview, qtbot):
         completionview.completion_item_focus(which)
 
 
+def test_completion_item_focus_fetch(completionview, qtbot):
+    """Test that on_next_prev_item moves the selection properly.
+
+    Args:
+        which: the direction in which to move the selection.
+        tree: Each list represents a completion category, with each string
+              being an item under that category.
+        expected: expected argument from on_selection_changed for each
+                  successive movement. None implies no signal should be
+                  emitted.
+    """
+    model = completionmodel.CompletionModel()
+    cat = mock.Mock(spec=listcategory.ListCategory)
+    cat.canFetchMore = lambda *_: True
+    cat.rowCount = lambda *_: 2
+    cat.fetchMore = mock.Mock()
+    model.add_category(cat)
```

Comment:
Looks like [this fails](https://travis-ci.org/qutebrowser/qutebrowser/jobs/256734083#L710) (but only with Qt 5.7.1?) because it wants `cat.layoutAboutToBeChanged`. But shouldn't `spec=listcategory.ListCategory` give you that anyways?